terraform/graph_config_node_variable.go

```diff
@@ -130,17 +130,22 @@ func (n *GraphNodeConfigVariable) hasDestroyEdgeInPath(opts *NoopOpts, vertex da
 	if vertex == nil {
 		vertex = opts.Vertex
 	}
+
+	var root graphNodeRoot
```

Comment:
I see how this works, but it feels like a bit of an odd way to test if a node is the root node. Maybe worth wrapping this up in a `IsGraphNodeRoot()` predicate stored near the `graphNodeRoot` impl?

Or here's another idea - IMHO this impl is a bit less confusing than the local var / `==` test:

```go
_, isRoot := v.(graphNodeRoot)
if isRoot || cv.DestroyEdgeInclude(v) { // ...
```

What do you think?